sycl/plugins/cuda/pi_cuda.cpp

```diff
@@ -4603,7 +4670,7 @@ pi_result cuda_piextUSMEnqueuePrefetch(pi_queue queue, const void *ptr,
                                        pi_event *event) {
   assert(queue != nullptr);
   assert(ptr != nullptr);
-  CUstream cuStream = queue->get();
+  CUstream cuStream = queue->get_mem();
```

Comment:
we need to enable disable using the alternative queue, ideally via some configuration option. 
```suggestion
  CUstream cuStream = UseAlternateForMem?(queue->get_mem()):(queue->get());
```

At least, include a preprocessor macro so that get_mem returns the original stream if the feature is disabled at compile time

Reply:
To me the tidiest option is what you suggest, have `get_mem()` return the default stream if its the only one that exists. However, I can see that this is also the more opaque approach.

Reply:
I've added an env var `PI_CUDA_USE_ALT_STREAM=1` to control the 2nd stream. The method `get_alt_stream()` will return the default (only) stream if this isn't set.